cep-20.2.md

```diff
@@ -0,0 +1,832 @@
+There are multiple types of test elements defined, such as the `command` test element, the `python` test element and the `downstream` test element.
+
+Before, the test section was written to a single folder (`info/test`). In the new format, we propose to write each test element to a separate folder
+(`info/tests/<index>`). This allows us to run each test element independently.
+
+```yaml
+tests: [TestElement]
+```
+
+#### Command test element
+
+The command test element renders to a single folder with a `test_time_dependencies.json` with two keys (`build` and `run`) that contain the raw "PackageSelector" strings.
+The `script` is rendered to a `script.json` that contains the `interpreter`, `env` and other keys (as defined in the `Script` section).
+Files are copied into the `info/tests/<index>` folder.
+
+```yaml
+# script to execute
+# reuse script definition from above
+script: string | [string] | Script
+# optional extra requirements
+requirements:
+  # extra requirements with build_platform architecture (emulators, ...)
+  build: [PackageSelector]
+  # extra run dependencies
+  run: [PackageSelector]
+
+# extra files to add to the package for the test
+files:
+  # files from $SRC_DIR
+  source: [glob]
+  # files from $RECIPE_DIR
+  recipe: [glob]
+```
+
+#### Python test element
+
+The python test element renders a `test_import.py` file that contains the imports to test.
+It also automatically runs the `pip check` command to check for missing dependencies.
+
+```yaml
+python:
+  # list of imports to try
+  imports: [string]
+  pip_check: bool  # defaults to true
+```
+
+#### Downstream test element
+
+The downstream test element renders to a `test_downstream.json` file that contains the `downstream` key with the raw "PackageSelector" string.
+
+```yaml
+downstream: PackageSelector
+```
+
+## Outputs section
+
+conda-build has very hard to understand behavior with multiple outputs. We propose some drastic simplifications.
+Each output in the new format has the same keys as a "top-level" recipe.
+
+Values from the top-level `build`, `source` and `about` sections are (deeply) merged into each of the outputs.
+
+The top-level package field is replaced by a top-level `recipe` field. The `version` from the top-level
+`recipe` is also merged into each output. The top-level name is ignored (but could be used, e.g. for
+the feedstock-name in the conda-forge case).
+
+```yaml
+# note: instead of `package`, it says `recipe` here
+recipe:
+  name: string # mostly ignored, could be used for feedstock name
+  version: string # merged into each output if not overwritten by output
+
+outputs:
+  - package:
+      name: string
+      version: string (defaults to top-level version)
+    build:
+      script: ...
+    requirements:
+      # same definition as top level
+
+    # same definitions as on top level, by default merged from outer recipe
```

Comment:
I'd define what merge means here:

* The output value overwrites the top-level value
* The two values are merged key-wise
* The two values are merged value-wise

Reply:
+1